packages/components/table/src/store/watcher.ts

```diff
@@ -118,15 +118,19 @@ function useWatcher<T>() {
       (column) =>
         column.type !== 'selection' && [true, 'left'].includes(column.fixed)
     )
-
+    // 找有fixed的selectColumn
+    const fixedSelectColumn = _columns.value.find(
+      (column) =>
+        column.type === 'selection' &&
+        [true, 'left', 'right'].includes(column.fixed)
+    )
```

Comment:
我觉得直接对所有列重新排序，把selection列放到最前面更加直观。
```suggestion
    const findSelectColumn = _columns.value.find(
      (column) => column.type === 'selection'
    )
    const sortedColumns = _columns.value.filter(
      (column) => column.type !== 'selection'
    )
    if (findSelectColumn) {
      sortedColumns.unshift(findSelectColumn)
    }
```

Reply:
我一开始也是这样想的，但是后面我觉得这样限制完以后，select永远在最前面排序了，可能用户想放在中间都没办法展示了

Reply:
其它的测试用例失败，可以提供一下复现代码嘛

Reply:
> select 放在中间看起来不是一个合理的需求，我在其他组件库中似乎也没见过这种用法。而且，目前的修改会导致其他测试用例失败，看起来也有问题。

view design组件库的table支持多个select， 导致其它测试用例修改失败，可以给一下复现代码嘛

Reply:
> 其它的测试用例失败，可以提供一下复现代码嘛

https://github.com/element-plus/element-plus/actions/runs/14775901698/job/41491609427?pr=20582
你可以本地执行`pnpm test`跑一下测试用例。

Reply:
> view design组件库的table支持多个select， 导致其它测试用例修改失败，可以给一下复现代码嘛

支持多个 select 的场景是什么？多个的话要怎么用？

Reply:
> > 其它的测试用例失败，可以提供一下复现代码嘛
> 
> https://github.com/element-plus/element-plus/actions/runs/14775901698/job/41491609427?pr=20582 你可以本地执行`pnpm test`跑一下测试用例。

test案例有问题,已修复